contracts/adapters/ibc/secret-ibc-hooks/src/contract.rs

```diff
@@ -0,0 +1,497 @@
+    sudo::{OsmosisSudoMsg as SudoMsg, SudoType},
+};
+use secret_toolkit::snip20;
+
+// const IBC_MSG_TRANSFER_TYPE_URL: &str = "/ibc.applications.transfer.v1.MsgTransfer";
+const REPLY_ID: u64 = 1;
+
+///////////////
+/// MIGRATE ///
+///////////////
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn migrate(deps: DepsMut, _env: Env, msg: MigrateMsg) -> ContractResult<Response> {
+    // Set contract version
+    // set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+    // Validate entry point contract address
+    let checked_entry_point_contract = ContractInfo {
+        address: deps
+            .api
+            .addr_validate(&msg.entry_point_contract.address.to_string())?,
+        code_hash: msg.entry_point_contract.code_hash,
+    };
+
+    // Store the entry point contract address
+    ENTRY_POINT_CONTRACT.save(deps.storage, &checked_entry_point_contract)?;
+
+    let checked_ics20_contract = ContractInfo {
+        address: deps
+            .api
+            .addr_validate(&msg.ics20_contract.address.to_string())?,
+        code_hash: msg.ics20_contract.code_hash,
+    };
+
+    ICS20_CONTRACT.save(deps.storage, &checked_ics20_contract)?;
+
+    Ok(Response::new()
+        .add_attribute("action", "migrate")
+        .add_attribute(
+            "entry_point_contract_address",
+            checked_entry_point_contract.address.to_string(),
+        )
+        .add_attribute(
+            "ics20_contract_address",
+            checked_ics20_contract.address.to_string(),
+        ))
+}
+
+///////////////////
+/// INSTANTIATE ///
+///////////////////
+
+// Contract name and version used for migration.
+/*
+const CONTRACT_NAME: &str = env!("CARGO_PKG_NAME");
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+*/
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: InstantiateMsg,
+) -> ContractResult<Response> {
+    // Set contract version
+
+    // Validate entry point contract address
+    let checked_entry_point_contract = ContractInfo {
+        address: deps
+            .api
+            .addr_validate(&msg.entry_point_contract.address.to_string())?,
+        code_hash: msg.entry_point_contract.code_hash,
+    };
+
+    // Store the entry point contract address
+    ENTRY_POINT_CONTRACT.save(deps.storage, &checked_entry_point_contract)?;
+
+    let checked_ics20_contract = ContractInfo {
+        address: deps
+            .api
+            .addr_validate(&msg.ics20_contract.address.to_string())?,
+        code_hash: msg.ics20_contract.code_hash,
+    };
+
+    ICS20_CONTRACT.save(deps.storage, &checked_ics20_contract)?;
+
+    Ok(Response::new()
+        .add_attribute("action", "instantiate")
+        .add_attribute(
+            "entry_point_contract_address",
+            checked_entry_point_contract.address.to_string(),
+        )
+        .add_attribute(
+            "ics20_contract_address",
```

Comment:
This ICS 20 contract address is a contract on secret that interact with the underlying IBC module, is that correct?

Reply:
yep, it handles wrapping and unwrapping for tokens going in and out of the network
https://github.com/scrtlabs/snip20-ics20